### PR TITLE
Add generate_averaged_model.py

### PR DESCRIPTION
**Pull request by yfyeung**
Generate the averaged model from exp-dir.

Usage:
(1) use the checkpoint exp_dir/epoch-iter.pt
./pruned_transducer_stateless7/generate_averaged_model.py \
    --epoch 28 \
    --avg 15 \
    --exp-dir ./pruned_transducer_stateless7/exp

(2) use the checkpoint exp_dir/checkpoint-iter.pt
./pruned_transducer_stateless7/generate_averaged_model.py \
    --iter 22000 \
    --avg 5 \
    --exp-dir ./pruned_transducer_stateless7/exp

### CODE REVIEW

File: egs/librispeech/ASR/pruned_transducer_stateless7/generate_averaged_model.py [hunk 1]
```diff
@@ -0,0 +1,199 @@
+#!/usr/bin/env python3
+#
+# Copyright 2021-2022 Xiaomi Corporation (Author: Fangjun Kuang,
```
Comment (csukuangfj) [inline egs/librispeech/ASR/pruned_transducer_stateless7/generate_averaged_model.py:3]: please change it to yours.

Comment (yfyeung) [reply egs/librispeech/ASR/pruned_transducer_stateless7/generate_averaged_model.py]: OK

**Review by csukuangfj — APPROVED** [review]
Thanks!

**Comment by danpovey** [comment]
I think it would be better if it had the --use-averaged-model option, so that it can do the averaged-over-checkpoints version like the decode script.

**Pull request merged**